include/swift/AST/Expr.h

```diff
@@ -2761,8 +2770,22 @@ class UnresolvedTypeConversionExpr : public ImplicitConversionExpr {
 class FunctionConversionExpr : public ImplicitConversionExpr {
 public:
   FunctionConversionExpr(Expr *subExpr, Type type)
-    : ImplicitConversionExpr(ExprKind::FunctionConversion, subExpr, type) {}
-  
+      : ImplicitConversionExpr(ExprKind::FunctionConversion, subExpr, type) {
+    FunctionConversionExprBits.Flattening = false;
+  }
+
+  /// Set whether this function conversion flattens an unapplied member
+  /// function.
+  void setFlattening() {
```

Comment:
Right, so that place can make the new Expr type instead. These really are not function conversions...